### PR TITLE
create docs site

### PR DESCRIPTION
**Pull request by gpwclark**
Generates static html doc site. Triggered from script (that will soon become a slosh script) that is run in github actions.

This includes an enumeration of every slosh form by section and any docs we specify.
In the docs we specify there is special support for code blocks marked as slosh code, e.g.
```slosh
(prn (+ 1 1))
```
will be preprocessed, that output, 2 will be attained and will be printed below the code block, e.g.
```
;; => 2
```

Adding a new doc and taking advantage of this is as easy as adding md files in to doc/src/ and referencing them in doc/src/SUMMARY.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -197,12 +197,19 @@ These forms (written in Rust but callable from Lisp) are supported.
...
 ### Generated documentation
+The [documentation site](https://sl-sh-dev.github.io/sl-sh/) is served from the `doc` directory
+based on the `doc/mk-site.sh` script referenced in the github action `.github/workflows/gh_pages.yml`.
+
+Documentation can be built and viewed locally as the documentation site is a static set of html files
+generated from the source code.
+
+All of this is covered in `doc/README.md`
+
+Must be compiled in lisp-test mode for appropriate documenation functions to be present (`cargo build --features list-test`).
```
Comment (sstanfield) [inline README.md:214]: I think you mean --features lisp-test not list-test.

File: doc/README.md [hunk 2]
```diff
@@ -13,8 +13,11 @@ rm -rf ./book/*
 ```
 
 
+ TODO PC need md files in a specific folder named after a section to be inlined within the section documetnation
+
 ## Important information
 
+0. Must be compiled in lisp-test mode for appropriate documenation functions to be present (`cargo build --features list-test`).
```
Comment (sstanfield) [inline doc/README.md:20]: lisp-test

File: slosh/Cargo.toml [hunk 3]
```diff
@@ -4,6 +4,9 @@ version = "0.10.0"
 edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
+[features]
```
Comment (sstanfield) [inline slosh/Cargo.toml:7]: Do we need "default=[]' here?  Want to make sure these dips are not only optional but are off by default.  Some anyway, regex will probably become default on when we wrap it with forms.

**Review by sstanfield — APPROVED** [review]
Couple minor comments to look at.  You have a clippy problem, the PR I just opened addresses it but it is an easy fix if yours goes first.

File: slosh/Cargo.toml (hunk 3 above)
Comment (gpwclark) [reply slosh/Cargo.toml:7]: yes. that's much more clear. good catch.

**Pull request merged**